arc/t.sys_maslload.c

```diff
@@ -148,7 +148,12 @@ static int ${te_prefix.result}MASL_load_file( const char * filepath, const struc
   /*
    * Open the named file for reading.
    */
-  if ( !strstr( filepath, ".masl" ) ) {
+  ${te_prefix.type}size_t len = ${te_prefix.result}strlen(filepath);
+  if ( !( '.' == filepath[len-5] &&
```

Comment:
There should be a check that the length of the string is >= 5. It should not ever matter because "the path will always be prepended by at least "models/", but still... my safe C buzzers are going off!